File: sdk/go/lima/images.go
package lima

import "context"

// ImageGeneration is the response from the image generation endpoint.
type ImageGeneration struct {
	Data []struct {
		URL     string `json:"url,omitempty"`
		B64JSON string `json:"b64_json,omitempty"`
	} `json:"data"`
}

// CreateImageRequest is the request body for image generation.
type CreateImageRequest struct {
	Model  string `json:"model"`
	Prompt string `json:"prompt"`
	Size   string `json:"size,omitempty"`
	N      int    `json:"n,omitempty"`
}

// ImagesService provides image generation.
type ImagesService struct {
	client *Client
}

// Generate creates an image from a prompt.
func (s *ImagesService) Generate(ctx context.Context, req CreateImageRequest) (*ImageGeneration, error) {
	var resp ImageGeneration
	if err := s.client.jsonRequest(ctx, "POST", "/v1/images/generations", req, &resp); err != nil {
		return nil, err
	}
	return &resp, nil
}

File: sdk/js/lima-sdk/src/resources/images.js
import { raiseForStatus } from "../client.js";

/** @typedef {import('../../types/index.d.ts').ImageGeneration} ImageGeneration */
/** @typedef {import('../client.js').LiMaClient} LiMaClient */

export class Images {
  /**
   * @param {LiMaClient} client
   */
  constructor(client) {
    this.client = client;
  }

  /**
   * @param {object} params
   * @param {string} params.model
   * @param {string} params.prompt
   * @param {string} [params.size]
   * @param {number} [params.n]
   * @param {Record<string, unknown>} [params.options]
   * @returns {Promise<ImageGeneration>}
   */
  async generate({ model, prompt, size = "1024x1024", n = 1, ...options }) {
    const response = await this.client.request("POST", "/v1/images/generations", {
      json: { model, prompt, size, n, ...options },
    });
    await raiseForStatus(response);
    return /** @type {ImageGeneration} */ (await response.json());
  }
}
